lib/SimpleSAML/Utils/HTTP.php

```diff
@@ -361,10 +362,14 @@ public static function fetch($url, $context = array(), $getHeaders = false)
         $config = \SimpleSAML_Configuration::getInstance();
 
         $proxy = $config->getString('proxy', null);
+        $proxy_auth = $config->getString('proxy.auth', null);
         if ($proxy !== null) {
             if (!isset($context['http']['proxy'])) {
                 $context['http']['proxy'] = $proxy;
             }
+            if (!empty($proxy_auth) ) {
```

Comment:
The comparison here should be:

```php
if ($proxy_auth !== null) {
```

Setting `proxy.auth` to `0` or `""` (empty string) would avoid triggering authorization. Actually, it's probably better to default it to false and compare this way:

```php
if ($proxy_auth !== false) {
```

Then we should state clearly in the documentation that setting `proxy.auth` to `false` will disable it.